.github/workflows/oaf-check.yml

```diff
@@ -11,18 +11,101 @@ jobs:
   oaf-up-to-date:
     name: Check for new OAF version
     runs-on: ubuntu-latest
-
+    outputs:
+      OAF_VERSION: ${{ steps.save-oaf-version.outputs.OAF_VERSION }}
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
           python-version: ${{ inputs.python-version }}
           cache: 'pip'
-          cache-dependency-path: 'requirements/dev.txt'
+          cache-dependency-path: 'requirements/*.txt'
       - name: Install dependencies
         run: pip install $(grep "pip-tools==" requirements/dev.txt)
       - name: Run compile dependencies
         run: ./bin/compile_dependencies.sh --upgrade-package open-api-framework
-
       - name: Check git diff
         run: git diff --exit-code -- requirements/*.txt
+      - name: Save OAF Version
+        # save the package version for branch names and commit messages
+        if: failure()
+        id: save-oaf-version
+        run: echo "OAF_VERSION=$(grep -Po '(?<=open-api-framework==)[^\n]*' requirements/dev.txt)" >> $GITHUB_OUTPUT
+
+  create-oaf-update-pr:
+    name: Create a pull requests to update OAF
+    runs-on: ubuntu-latest
+    needs: [oaf-up-to-date]
+    # only run if not up to date and version is provided
+    if: failure()
+    env:
+      OAF_VERSION: ${{needs.oaf-up-to-date.outputs.OAF_VERSION}}
+      OAF_BRANCH: update/open-api-framework-${{needs.oaf-up-to-date.outputs.OAF_VERSION}}
+      ACTION_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+    steps:
+      # Setup Git
+      - uses: actions/checkout@v4
+      - name: Set up Git credentials
+        run: |
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+      - name: Checkout or Create Branch
+        # Checkout of create new branch if it already exists
+        run: |
+          git fetch
+          git switch $OAF_BRANCH || \
```

Comment:
I think only using `git switch --force-create` here would make sense (to always start with a clean slate), as we never know what state the existing branch might be in (it is assumed here that the correct changes are in there already).

Reply:
It always runs compile dependencies so I don't think it matters too much. 

Force checkout is simpler though and I want to change it to `git diff local origin` so it shouldn't make a difference.
Also I always force push one commit so yeah the rebase makes no sense and I am just complicating things.